hack/install-vm.sh

```diff
@@ -0,0 +1,407 @@
+# limitations under the License.
+​
+INSTALL_PATH="/usr/sbin"
+ANTREA_AGENT="antrea-agent"
+AGENT_LOG_DIR="/var/log/antrea"
+AGENT_CONF_PATH="/etc/antrea"
+OVS_BRIDGE="br-int"
+OVS_VSWITCHD="ovs-vswitchd.service"
+ANTREA_DOCKER_IMAGE="projects.registry.vmware.com/antrea/antrea-ubuntu:v1.9.0"
+OVS_BIN="start_ovs"
+ANTREA_AGENT_BIN="antrea-agent"
+ANTREA_AGENT_KUBECONFIG="antrea-agent.kubeconfig"
+ANTREA_AGENT_ANTREA_KUBECONFIG="antrea-agent.antrea.kubeconfig"
+
+_usage="Usage: $0 [--ns <Namespace>] [--bin <AntreaAgentSavePath>] [--config <AgentConfigSavePath>] [--kubeconfig <KubeconfigSavePath>] [--antrea-kubeconfig <AntreaKubeconfigSavePath>] [--nodename <ExternalNodeName>] [--ovs-bridge <OVSBridgeName>] [--install-docker <InstallDocker>] [--validate-ovs] [--help|-h]
+        --ns                          Namespace to be used by the antrea-agent
+        --bin                         Path of the antrea-agent binary
+        --config                      Path of the antrea-agent configuration file
+        --kubeconfig                  Path of the kubeconfig to access K8s API Server
+        --antrea-kubeconfig           Path of the kubeconfig to access Antrea API Server
+        --nodename                    ExternalNode name to be used by the antrea-agent
+        --ovs-bridge                  Specify the OVS bridge name
+        --validate-ovs                Validate OVS configuration and performs cleanup when any error is detected.
+        --install-docker              Specify true or false to install or skip the docker installations.
+        --help, -h                    Print this message and exit
+​
+Please run the script as sudo user"
+
+VALIDATE_OVS_CONFIG=false
+INSTALL_DOCKER=true
+NODE_NAME="$(hostname)"
+
+declare -a SUPPORTED_OS=("Ubuntu 18.04" "Ubuntu 20.04" "Red Hat Enterprise Linux 8.4")
+
+set -eo pipefail
+​
+function echoerr {
+    >&2 echo "$@"
+}
+​
+function print_usage {
+    echoerr "$_usage"
+}
+​
+function print_help {
+    echoerr "Try '$0 --help' for more information."
+}
+​
+check_ovs_config_and_cleanup() {
+    if [ "$VALIDATE_OVS_CONFIG" = true ] ; then
+      bridges=$(ovs-vsctl list-br)
+      for br in $bridges; do
+          if [ "$br" != "$OVS_BRIDGE" ] ; then
+              continue
+          fi
+          # Check if any of the interface is in error state.
+          ports=$(ovs-vsctl list-ports $OVS_BRIDGE)
+          for port in $ports; do
+              output=$(ovs-vsctl --no-headings --columns=error list interface "$port")
+              if [ "$output" != '[]' ] ; then
+                  echoerr "Error while listing interface $port, deleting bridge $OVS_BRIDGE"
+                  ovs-vsctl del-br "$OVS_BRIDGE"
+                  break
+              fi
+          done
+          exit 0
+      done
+      exit 0
+    fi
+}
+
+validate_mandatory_arguments() {
+    if [ -z "$CONFIG_PATH" ] || [ -z "$KUBECONFIG" ] || [ -z "$ANTREA_KUBECONFIG" ] ; then
```

Comment:
For native installation, --bin argument is must. 
For docker way its not required.